elasticdl/python/master/master.py

```diff
@@ -68,8 +68,9 @@ def _maybe_create_shards(data_origin):
 class Master(object):
     def __init__(self, args):
         self.logger = get_logger("master", level=args.log_level.upper())
-        self.checkpoint_output_path = args.output
+
         self.num_ps_pods = args.num_ps_pods
+        self.checkpoint_output_path = args.checkpoint_dir
```

Comment:
It seems a little confused that we save model in two formats, one is `checkpoint`, another is `tf.saved_model`. Would it be convenient if we change the save checkpoint file in the same format as `tf.saved_model`?

Reply:
Checkpoint is necessary during training, but it only stores the parameter value including optimizer slot value, it's used to restore the training process if preempted.

SavedModel contains both variable and model graph definition. It's used for serving and doesn't contain the optimizer state. What's more, it's a heavier operation.

Reply:
Saving checkpoint now ignores many optimizer states. 

Yes, I known there are differences between checkpoint and SavedModel. But it seems no need to use two logic and format to save them. If we can save them in the same format, it would reuse many code and make our code base more clean. For example, save checkpoint in the same logic and format as used for savedModel, but only save variables and do not save graph definition. Any way, you can finish the project of serving first and then consider cleaning code.